centml/cli/cluster.py

```diff
@@ -20,10 +21,46 @@ def convert(self, value, param, ctx):
             return None  # to avoid warning from lint for inconsistent return statements
 
 
-hw_to_id_map = {"small": 1000, "medium": 1001, "large": 1002}
-id_to_hw_map = {v: k for k, v in hw_to_id_map.items()}
+# Cache hardware instances
+_hw_id_map_cache = None
+# Declare global variables with type annotations
+hw_to_id_map: Dict[str, int] = {}
+id_to_hw_map: Dict[str, str] = {}
 
 
+def fetch_hardware_mappings():
+    global _hw_id_map_cache
+    if _hw_id_map_cache is None:
+        _hw_id_map_cache = api.get_hardware_instances()
+    return _hw_id_map_cache
+
+
+def get_hw_to_id_map():
+    global hw_to_id_map, id_to_hw_map
+
+    response = fetch_hardware_mappings()
+
+    # Handle empty response
+    if not response:
+        click.echo("Failed to fetch hardware instances. Please try again.")
+        return hw_to_id_map, id_to_hw_map
+
+    # Convert to list of dictionaries
+    instances = [item.to_dict() for item in response]
+
+    # Clear existing mappings
+    hw_to_id_map = {}
+    id_to_hw_map = {}
+
+    for item in instances:
+        hw_to_id_map[item["name"]] = item["id"]
+        id_to_hw_map[str(item["id"])] = item["name"]
+    return hw_to_id_map, id_to_hw_map
+
+
+# Fetch the hardware instance mappings
+hw_to_id_map, id_to_hw_map = get_hw_to_id_map()
```

Comment:
if you are always running this anyway, the above logic of caching and checking the cache is pointless for CLI for the reasons described above.

Reply:
you can just make the API call inside get_hw_to_id_map() and return both mappings.